### PR TITLE
[MERGE] rating: cleanup customer rating flow and views

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
PURPOSE

Cleanup code and flow of customer rating: routes, rating_apply, rating and
message creation and update. Cleanup some view definitions.

SPECIFICATIONS

Cleanup code about rating controllers and rating_apply. Notably correctly
link a message to a rating once a feedback is posted, whatever the flow.

Update ``rating_apply`` to use either a token, either an existing rating to
udpate it and post a message. Always link the message and the rating to
be sure DB data are coherent.

This requires an update in several demo data to correctly set tokens on
ratings and use it in calls to ``rating_apply`` done in xml files.

Extract default subtype computation of rating_apply in a sub-method to allow
setting this parameter without having to deal with ``rating_apply`` details.

CODE CLEANUP

Split behavior that is generic but actually used only in project about stage
update based on rating. Move it directly in project. As it is used only once
and as generic approach is hardcoded based on fields names better keep it
localized.

VIEW CLEANUP

Move some ratings views specific to some modules directly into rating to
offer some different views based on usage (mainly text / image of rating).

Task-2812665 (Rating: Cleanup rating flow code)

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
@IT-Ideas @xavierbol FYI (still a bit WIP)

**Review by std-odoo — COMMENTED** [review]
Validation :)

File: addons/rating/models/rating_mixin.py [hunk 1]
```diff
@@ -173,40 +173,57 @@ def rating_send_request(self, template, lang=False, subtype_id=False, force_send
...
+            # already been notified (new message will own the rating)
+            if not rating.message_id or rating.message_id.notification_ids:
                 self.message_post(
-                    body="<img src='/rating/static/src/img/rating_%s.png' alt=':%s/5' style='width:18px;height:18px;float:left;margin-right: 5px;'/>%s"
-                    % (rate, rate, feedback),
-                    subtype_xmlid=subtype_xmlid or "mail.mt_comment",
-                    author_id=rating.partner_id and rating.partner_id.id or None  # None will set the default author in mail_thread.py
+                    author_id=rating.partner_id.id or None,  # None will set the default author in mail_thread.py
+                    body="<img src='%s' alt=':%s/5' style='width:18px;height:18px;float:left;margin-right: 5px;'/>%s"
+                    % (rating.rating_image_url, rate, feedback),
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/rating/models/rating_mixin.py:223]: As this is plain text transformed into html using the tool, this should not be escaped again, as discussed on discord :) .

**Review by tde-banana-odoo — COMMENTED** [review]
Validation done / answered :)

**Comment by std-odoo** [comment]
LGTM :)

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**